drivers/pwm/pwm_max31790.c

```diff
@@ -0,0 +1,685 @@
+	MAX37190_SET_VALUE(target, value, MAX37190_FANXCONFIGURATION_TACH_POS,                     \
+			   MAX37190_FANXCONFIGURATION_TACH_LENGTH)
+#define MAX37190_FANXDYNAMICS_SPEEDRANGE_LENGTH 3
+#define MAX37190_FANXDYNAMICS_SPEEDRANGE_POS	5
+#define MAX37190_FANXDYNAMICS_SPEEDRANGE_GET(value)                                                \
+	MAX37190_GET_VALUE(value, MAX37190_FANXDYNAMICS_SPEEDRANGE_POS,                            \
+			   MAX37190_FANXDYNAMICS_SPEEDRANGE_LENGTH)
+#define MAX37190_FANXDYNAMICS_SPEEDRANGE_SET(target, value)                                        \
+	MAX37190_SET_VALUE(target, value, MAX37190_FANXDYNAMICS_SPEEDRANGE_POS,                    \
+			   MAX37190_FANXDYNAMICS_SPEEDRANGE_LENGTH)
+#define MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_LENGTH 3
+#define MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_POS    2
+#define MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_GET(value)                                           \
+	MAX37190_GET_VALUE(value, MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_POS,                       \
+			   MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_LENGTH)
+#define MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_SET(target, value)                                   \
+	MAX37190_SET_VALUE(target, value, MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_POS,               \
+			   MAX37190_FANXDYNAMICS_PWMRATEOFCHANGE_LENGTH)
+#define MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_LENGTH 1
+#define MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_POS    1
+#define MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_GET(value)                                    \
+	MAX37190_GET_VALUE(value, MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_POS,                \
+			   MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_LENGTH)
+#define MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_SET(target, value)                            \
+	MAX37190_SET_VALUE(target, value, MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_POS,        \
+			   MAX37190_FANXDYNAMICS_ASYMMETRICRATEOFCHANGE_LENGTH)
+
+static int max31790_read_register(const struct device *dev, enum max31790_register address,
+				  uint8_t *value)
+{
+	const struct max31790_config *config = dev->config;
+	int result = i2c_reg_read_byte_dt(&config->i2c, address, value);
+
+	LOG_DBG("read value 0x%02X from register 0x%02X", *value, address);
+
+	if (result != 0) {
+		LOG_ERR("unable to read register 0x%02X, error %i", address, result);
+	}
+
+	return result;
+}
+
+static int max31790_write_register(const struct device *dev, enum max31790_register address,
+				   uint8_t value)
+{
+	const struct max31790_config *config = dev->config;
+
+	LOG_DBG("writing value 0x%02X to register 0x%02X", value, address);
+	int result = i2c_reg_write_byte_dt(&config->i2c, address, value);
+
+	if (result != 0) {
+		LOG_ERR("unable to write register 0x%02X, error %i", address, result);
+	}
+
+	return result;
+}
+
+static bool max31790_convert_pwm_frequency_into_hz(uint16_t *result, uint8_t pwm_frequency)
+{
+	switch (pwm_frequency) {
+	case 0b0000:
```

Comment:
Why the binary representation? Curiosity mor than anything else, it's just counting...

Reply:
This is the exact same mapping as in the datasheet, therefore this format.

Reply:
I cannot find binary constants in C99 standard.

Reply:
I cannot find in the requirements for zephyr sticking to C99:
https://docs.zephyrproject.org/latest/develop/languages/c/index.html

But it looks like the binary literals are not C standard at all, just a common proprietary extension. Therefore I will replace them, because not using extensions seems to be reasonable to me.

Reply:
I replaced all the binary literals, not only those from this switch statement.

Reply:
It's an interesting topic though, filed https://github.com/zephyrproject-rtos/zephyr/issues/60564 to ask for more opinions.